terraform-module/modules/frontend-spa-cdn/main.tf

```
@@ -161,5 +195,73 @@ resource "aws_cloudfront_response_headers_policy" "default_behaviour_headers_pol
       override = true
       value    = var.is_robots_indexing_allowed && var.env == "production" ? "all" : "noindex, nofollow"
     }
+
+    // If page is opened via window.open() or by navigating to a new page
+    // it will be opened in a new browsing context group
+    // unless opener has the same origin as the target page.
+    // This prevents sharing the browsing context group with potentially malicious opener page
+    items {
+      header   = "Cross-Origin-Opener-Policy"
+      override = true
+      value    = "same-origin-allow-popups"
+    }
+  }
+}
+
+// This is custom headers policy for /loading.html page
+// This page is used as an intermediate page for 3rd party integration connections
+// And it should have ability to share same browsing context group with the opener page
+resource "aws_cloudfront_response_headers_policy" "loading_integration_behaviour_headers_policy" {
+  name = "${var.app_name}-loading-integration-headers-policy"
+  security_headers_config {
+    content_type_options {
+      override = true
+    }
+    frame_options {
+      frame_option = var.block_iframes ? "DENY" : "SAMEORIGIN"
+      override     = true
+    }
+    referrer_policy {
+      referrer_policy = "same-origin"
+      override        = true
+    }
+    xss_protection {
+      mode_block = true
+      protection = true
+      override   = true
+    }
+    strict_transport_security {
+      access_control_max_age_sec = "63072000"
+      include_subdomains         = true
+      preload                    = true
+      override                   = true
+    }
+  }
+
+  custom_headers_config {
+
+    // Adds cache control HTTP headers to the response to remove any caching
+    // Since we're only handling skeleton HTML files in this behaviour, disabling
+    // caching has little performance overhead. All static assets are cached aggressively
+    // in another behaviour.
+    items {
+      header   = "Cache-Control"
+      override = true
+      value    = "max-age=0,no-cache,no-store,must-revalidate"
+    }
+
+    // Adds robots tag HTTP header to the response to prevent indexing by bots (unless in production)
+    items {
+      header   = "X-Robots-Tag"
+      override = true
+      value    = var.is_robots_indexing_allowed && var.env == "production" ? "all" : "noindex, nofollow"
+    }
```

Why do we need to allow bots indexing for loading.html in prod?

---

I don't expect any problems having it indexed but also can't see any reason for allowing it - should we just set it to `noindex,nofollow` ?

```suggestion
    // Adds robots tag HTTP header to the response to prevent indexing by bots (unless in production)
    items {
      header   = "X-Robots-Tag"
      override = true
      value    =  "noindex, nofollow"
    }
```

---

We do not. I just copy pasted this config from the default settings. Will remove it